### PR TITLE
AWS_IoT: Add QEMU support

### PR DESCRIPTION
**Pull request by simensrostad**
This patch adds:
 - Internal credential provisioning and TLS caching in the AWS IoT library.
 - QEMU support in the AWS IoT library and sample.

PR depends on changes in #2751.

### CODE REVIEW

File: subsys/net/lib/aws_iot/src/aws_iot.c [hunk 1]
```diff
@@ -156,7 +160,7 @@ static int connect_error_translate(const int err)
 	default:
 		LOG_ERR("AWS broker connect failed %d", err);
 		return CLOUD_CONNECT_RES_ERR_MISC;
-#endif
+#endif /* !defined(CONFIG_CLOUD_LIBRARY) */
```
Comment (jtguggedal) [inline subsys/net/lib/aws_iot/src/aws_iot.c:163]: This should probably be `!defined(CONFIG_CLOUD_API)`

File: subsys/net/lib/aws_iot/src/aws_iot.c [hunk 2]
```diff
@@ -266,7 +270,7 @@ static void aws_iot_notify_event(const struct aws_iot_evt *aws_iot_evt)
 	} else {
 		LOG_ERR("Library event handler not registered, or empty event");
 	}
-#endif
+#endif /* !defined(CONFIG_CLOUD_LIBRARY) */
```
Comment (jtguggedal) [inline subsys/net/lib/aws_iot/src/aws_iot.c:273]: Same as above, `CONFIG_CLOUD_API`

File: subsys/net/lib/aws_iot/src/aws_iot.c [hunk 3]
```diff
@@ -9,6 +9,10 @@
 #include <net/aws_fota.h>
 #endif
 
+#if defined(CONFIG_BOARD_QEMU_X86)
```
Comment (jtguggedal) [inline subsys/net/lib/aws_iot/src/aws_iot.c:12]: `certificates_provision()` is conditionally compiled with `!defined(CONFIG_BSD_LIBRARY)`, and I think that would make sense here as well.

**Review by jtguggedal — COMMENTED** [review]
Looks good, just a couple of minor issues spotted

**Comment by simensrostad** [comment]
@rlubos @jtguggedal Addressed your review points. Thank you.

**Review by rlubos — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**